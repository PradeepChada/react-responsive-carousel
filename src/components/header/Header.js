import * as React from 'react';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import { useHistory } from 'react-router-dom';
import useScrollTrigger from '@mui/material/useScrollTrigger';
import AppLogo from './../../assets/images/logo.svg';
import HomeIcon from './../../assets/icons/home.svg';
import { Logo, StyledAppBar, TakeCheckout } from './Header.styles';
import { showCheckoutHeader } from '../../utils/skuHelpers';
import PropTypes from 'prop-types';

function ElevationScroll(props) {
  const { children, window } = props;
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 0,
    target: window ? window() : undefined,
  });

  return React.cloneElement(children, {
    elevation: trigger ? 4 : 0,
  });
}

ElevationScroll.propTypes = {
  children: PropTypes.element.isRequired,
  window: PropTypes.func,
};

const Header = () => {
  const history = useHistory();
  return (
    <ElevationScroll>
      <StyledAppBar position='sticky' color='inherit'>
        <Toolbar sx={{ alignItems: 'center' }}>
          <IconButton
            size='large'
            edge='start'
            color='inherit'
            aria-label='open drawer'
            className='home-icon'
            sx={{ ml: 2 }}
            onClick={() => history.push('/')}
            data-testid='home-icon'
          >
            <img src={HomeIcon} alt='Home' />
          </IconButton>
          <Box sx={{ flexGrow: 1, justifyContent: 'center', display: 'flex' }}>
            {showCheckoutHeader(history.location.pathname) ? (
              <TakeCheckout>Take Checkout</TakeCheckout>
            ) : (
              <Logo src={AppLogo} alt='The Container Store' />
            )}
          </Box>
        </Toolbar>
      </StyledAppBar>
    </ElevationScroll>
  );
};

export default Header;
