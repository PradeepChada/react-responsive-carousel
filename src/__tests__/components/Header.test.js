import { render, screen, fireEvent } from '@testing-library/react';
import Header from '../../components/header/Header';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('Testing Header component', () => {
  test('should render a snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test('should render Home Icon and Logo', () => {
    render(<Header />);
    const logo = screen.getByAltText('The Container Store');
    expect(logo).toBeInTheDocument();
    const homeIcon = screen.getByTestId('home-icon');
    expect(homeIcon).toBeInTheDocument();
  });

  test('should navidate to homepage when click on logo', () => {
    render(<Header />);
    const homeIcon = screen.getByTestId('home-icon');
    fireEvent.click(homeIcon);
  });
});
