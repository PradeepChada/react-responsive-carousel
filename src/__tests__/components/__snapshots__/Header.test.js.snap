// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component should render a snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic ForwardRef(AppBar)-root-1 css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1q8k0pk-MuiToolbar-root"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge ForwardRef(IconButton)-root-2 css-hu6rs9-MuiButtonBase-root-MuiIconButton-root"
          data-testid="home-icon"
          tabindex="0"
          type="button"
        >
          <img
            alt="Home"
            src="home.svg"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root css-1l8q18s"
        >
          <img
            alt="The Container Store"
            class="img-root-3"
            src="logo.svg"
          />
        </div>
      </div>
    </header>
  </div>
</div>
`;
